part-spec/common/pinSpec.json

```diff
@@ -1,253 +1,275 @@
-                "type": "array",
-                "items": {
-                    "$ref": "#/$defs/externalComponents"
-                }
-            },
-            "pinPaths": {
-                "description": "information on pin paths - pins associated with each component in a multi-component part (such as A1,Y1 and A2,Y2)",
-                "$ref": "../common/pinPaths.json#/pinPaths"
-            }
-        },
-        "additionalProperties": false
+  "$id": "https://github.com/edatasheets/digital-datasheets/blob/main/part-spec/common/pinSpec.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "specification of pins",
+  "pinSpec": {
+    "type": "object",
+    "required": [
+      "terminalIdentifier",
+      "name"
+    ],
+    "properties": {
+      "terminalIdentifier": {
+        "description": "pin or ball number as defined by datasheet",
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      },
+      "name": {
+        "description": "name given to the signal appearing at the terminal of a component",
+        "type": "string"
+      },
+      "standardizedName": {
+        "description": "standard name of pin",
+        "type": "string",
+        "enum": [
+          "drain",
+          "gate",
+          "source",
+          "anode",
+          "cathode",
+          "vdd",
+          "vss"
+        ]
+      },
+      "description": {
+        "description": "description of the signal appearing at the terminal of an electric/electronic component",
+        "type": "string"
+      },
+      "numberOfSupportedFunctions": {
+        "description": "the total number of functions supported by this pin",
+        "type": "number"
+      },
+      "pinProperties": {
+        "description": "list of properties for each pin",
+        "type": "array",
+        "items": {
+          "$ref": "#/$defs/pinProperties"
+        }
+      },
+      "functionProperties": {
+        "description": "list of properties for each pin function configuration",
+        "type": "array",
+        "items": {
+          "$ref": "#/$defs/functionProperties"
+        }
+      },
+      "pinPaths": {
+        "description": "information on pin paths - pins associated with each component in a multi-component part (such as A1,Y1 and A2,Y2)",
+        "$ref": "../common/pinPaths.json#/pinPaths"
+      }
+    },
+    "additionalProperties": false
+  },
+  "$defs": {
+    "functionProperties": {
+      "type": "object",
+      "properties": {
+        "perFunctionName": {
+          "description": "name of the function of a pin",
+          "examples": [
+            "UART5_TX",
+            "SPI3_COPI",
+            "PC12"
+          ],
+          "type": "string"
+        },
+        "perFunctionProperties": {
+          "description": "list of pin properties that change based on the pin function configuration",
```

Comment:
Also I don't think this is an array, I think it's just the object with whatever properties that are different